data-prepper-plugins/aggregate-processor/src/test/java/com/amazon/dataprepper/plugins/processor/aggregate/actions/CombineAggregateActionTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.aggregate.actions;
+
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.event.JacksonEvent;
+import com.amazon.dataprepper.plugins.processor.aggregate.AggregateAction;
+import com.amazon.dataprepper.plugins.processor.aggregate.AggregateActionResponse;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class CombineAggregateActionTest {
+    private AggregateAction combineAggregateAction;
+    private List<Event> events;
+    private List<Map<String, Object>> eventMaps;
+
+    @BeforeEach
+    void setup() {
+        events = new ArrayList<>();
+        eventMaps = new ArrayList<>();
+        combineAggregateAction = new CombineAggregateAction();
```

Comment:
I recommend that you use the `createObjectUnderTest()` pattern. This helps prevent calling methods in the setup method. It is more of a long-term maintainability benefit as the code changes hands over time.